src/building/how-to-build-and-run.md

```diff
@@ -223,6 +220,52 @@ fall back to using `cargo` from the installed `nightly`, `beta`, or `stable` too
 `rustup install nightly` if you haven't already.  See the
 [rustup documentation on custom toolchains](https://rust-lang.github.io/rustup/concepts/toolchains.html#custom-toolchains).
 
+## Building targets for cross-compilation
+
+To produce a compiler that can cross-compile for other targets,
+pass the `target` flag to `x.py build`.
+For example, if your host platform is `x86_64-unknown-linux-gnu`
+and your cross-compilation target is `wasm32-wasi`, you can build with:
+
+```bash
+./x.py build --target x86_64-unknown-linux-gnu --target wasm32-wasi
+```
+
+Note that you can pass multiple `target` flags in order to build support for multiple targets.
+Furthermore, if you want the resulting compiler to be able to build crates that
+involve proc macros or build scripts, you must be sure to explicitly build target support for the
+host platform (in this case, `x86_64-unknown-linux-gnu`).
+
+Note that building support for cross-compilation targets will probably require you to manually
+install any necessary target-specific packages, as well as provide additional configuration
+in `config.toml`.
+For example, attempting to build for the `wasm32-wasi` target requires providing a WASI sysroot
+and specifying the path to it via the target-specific `wasi-root` config key.
+Targets also may require additional configuration, for example `wasm32-wasi` also requires LLD.
+As a result, building support for `wasm32-wasi` requires `config.toml` to contain the following:
+
+```toml
+[rust]
+lld = true
+
+[target.wasm32-wasi]
+wasi-root = "/path/to/wasi/sysroot"
+```
```

Comment:
this specifically seems like it should go in https://doc.rust-lang.org/rustc/platform-support.html like @ehuss suggested, and we can link there from the guide. everything else looks good :)

Reply:
I'm happy to add a new page there for wasm32-wasi, however, I'm not qualified to fill out all the fields in the platform docs template (I don't know who the maintainers are, I don't know the platform requirements...). Is it okay if I just leave those fields blank and only fill out the "Building" section?

Reply:
Actually, for now I can just have this document link to any of the other platform pages that already exist. I can work on getting a wasm32-wasi page added separately.

Reply:
> Is it okay if I just leave those fields blank and only fill out the "Building" section?

Yes, I think that's fine :) incremental progress is better than no progress.